include/aos/common/pkcs11utils.hpp

```diff
@@ -0,0 +1,155 @@
+/*
+ * Copyright (C) 2023 Renesas Electronics Corporation.
+ * Copyright (C) 2023 EPAM Systems, Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef AOS_IAM_MODULES_PKCS11_UTILS_HPP_
+#define AOS_IAM_MODULES_PKCS11_UTILS_HPP_
+
+#include "aos/common/pkcs11.hpp"
+#include "aos/common/tools/memory.hpp"
+
+namespace aos {
+namespace pkcs11 {
+
+/**
+ * A Curve represents a short-form Weierstrass curve.
+ */
+enum class EllipticCurve {
+    /**
+     * A Curve which implements NIST P-384 (FIPS 186-3, section D.2.4), also known as secp384r1
+     **/
+    eP384
+};
+
+/**
+ * Crypto private key extended with object handle.
+ */
+class PrivateKey : public crypto::PrivateKey {
+public:
+    virtual ObjectHandle getHandle() = 0;
+};
+
+/**
+ * A helper class for a base Cryptoki API.
+ */
+class Utils {
+public:
+    /**
+     * Creates an object instance.
+     *
+     * @param session session context.
+     * @param allocator allocator for token/session objects.
+     */
+    Utils(pkcs11::SessionContext& session, Allocator& allocator)
```

Comment:
Unclear why we should pass the allocator for utils. Could utils have own private allocator? I need usage example.